BasicTypesProxy.h

```diff
@@ -312,7 +316,7 @@ namespace caf
       if(fType != kFlat || TreeHasLeaf(fTree, IndexField())){
         // Regular out-of-line array, handled the same as a vector.
         EnsureIdxP();
-        fElems[i] = new Proxy<T>(fTree, Subscript(i), fIdxP ? fIdxP->GetValue() : 0, i);
+        fElems[i] = new Proxy<T>(fTree, Subscript(i), fIdx, i);
```

Comment:
@jedori0228 As I said, I don't intuit exactly how this code works, but I am trying to rebase my extensive rewrite of SRProxy on top of these fixes and it stood out to me that this usage of `fIdx` doesn't get set with something like:


```
EnsureIdxP();
       if(fIdxP) fIdx = *fIdxP; // store into an actual value we can point to

       if(!fElems[i]) fElems[i] = new Proxy<T>(fTree, Subscript(i), fIdx, i);
 ```
 
 as it does in other places, can you confirm that you don't need to set fIdx for this codepath and it will be correctly set elsewhere?

Reply:
@luketpickering, if I understand the question correctly, in `EnsureElem(int i)` defined in `class Proxy<T[N]>` (Proxy for array), we don't have such lines that set `fIdx`. But for `EnsureLongEnough(size_t i)` in `Proxy<std::vector<T>>` (Proxy of vector), we have `if(fIdxP) fIdx = *fIdxP;`. If so, I have to figure out how to test arrays. I only have a setup that tests vectors..

Reply:
@luketpickering you gave approval already, but since this came up after approval, are you still happy to proceed? 
we can open up an issue to follow through with this but in the interests of making progress I'd be happy to make this resolved , make a new version of srproxy, follow this thread in an issue. Thoughts @cafana/stakeholders ?

Reply:
If this is not a question of immediate consequence to the functionality of proxy, sbn would prefer we move to merge this and resolve the question offline.

Reply:
agreed from NOvA also.  (I also have another proposed change to `SRProxy`, see #7,  that I'd like to bundle together with this one.)

Reply:
I have opened issue #8 to track this remaining check.
Given that there seems to be consensus that we should proceed regardless, I am taking the liberty of resolving this conversation.